pkg/render/render.go

```diff
@@ -0,0 +1,108 @@
+package render
+
+import (
+	"bytes"
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+	"text/template"
+
+	"github.com/Masterminds/sprig"
+	"github.com/pkg/errors"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/util/yaml"
+)
+
+type RenderData struct {
+	Funcs template.FuncMap
+	Data  map[string]interface{}
+}
+
+func MakeRenderData() RenderData {
+	return RenderData{
+		Funcs: template.FuncMap{},
+		Data:  map[string]interface{}{},
+	}
+}
+
+// RenderDir will render all manifests in a directory, descending in to subdirectories
+// It will perform template substitutions based on the data supplied by the RenderData
+func RenderDir(manifestDir string, d *RenderData) ([]*unstructured.Unstructured, error) {
+	out := []*unstructured.Unstructured{}
+
+	if err := filepath.Walk(manifestDir, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+		if info.IsDir() {
+			return nil
+		}
+
+		// Skip non-manifest files
+		if !(strings.HasSuffix(path, ".yml") || strings.HasSuffix(path, ".yaml") || strings.HasSuffix(path, ".json")) {
+			return nil
+		}
+
+		objs, err := RenderTemplate(path, d)
+		if err != nil {
+			return err
+		}
+		out = append(out, objs...)
+		return nil
+	}); err != nil {
+		return nil, errors.Wrap(err, "error rendering manifests")
+	}
+
+	return out, nil
+}
+
+// RenderTemplate reads, renders, and attempts to parse a yaml or
+// json file representing one or more k8s api objects
+func RenderTemplate(path string, d *RenderData) ([]*unstructured.Unstructured, error) {
```

Comment:
Do we really need the Funcs ? at render-manifests.go it was enough with ParseGlob, and those templates are not going to change, but maybe you know some tunning needed at manifests for openshift.